crispy_forms/tests/test_layout.py

```diff
@@ -598,3 +599,54 @@ def test_update_attributes_class():
     form.helper["password1"].update_attributes(css_class="hello2")
     html = render_crispy_form(form)
     assert html.count(' class="hello hello2 textinput') == 1
+
+
+@only_bootstrap4
+def test_file_field():
+    form = FileForm()
+    form.helper = FormHelper()
+    form.helper.layout = Layout("clearable_file")
+    html = render_crispy_form(form)
+    assert contains_partial(
```

Comment:
We don't need to worry about whitespace with contains_partial. I did try using Django's `assertinhtml` but ran into trouble with this. I think it's a unit test thing 🤷‍♂ . Anyway, we have contains partial which is similar.